app/controllers/admin/books_controller.rb

```diff
@@ -2,7 +2,12 @@ class Admin::BooksController < AdminController
   before_action :load_book, except: %i(index new create)
 
   def index
-    @pagy, @books = pagy Book.recent_add, items: Settings.page_items_10
+    if params[:term]
+      @pagy, @books = pagy Book.search(params[:term]),
+                           items: Settings.page_items_10
+    else
+      @pagy, @books = pagy Book.recent_add, items: Settings.page_items_10
+    end
```

Comment:
```suggestion
    @pagy, @books = pagy Book.search(params[:term]),
                           items: Settings.page_items_10
```

xử lý check param ở scope xử ý đi em
ko cần handle ở controller đâu

`scope :search, ->(term){where("name LIKE ?", "%#{term}%") if term.present?}`

Ngoài ra, nếu em muốn xử lý ở controller thì chị suggest cách để tránh lặp code hơn nè 

```suggestion
    @books = if params[:term]
      Book.search(params[:term])
    else
      Book
    end.recent_add
    
    @pagy, @books = pagy @books, items: Settings.page_items_10
```